lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -209,24 +214,36 @@ def self.is_firebasecmd_supported?(cmd)
         true
       end
 
+      def self.post_notes(app_id, release_id, release_notes)
+        payload = { releaseNotes: { releaseNotes: release_notes } }
+        if release_notes.nil? || release_notes.empty?
+          UI.message("No release notes passed in. Skipping this step.")
+          return
+        end
+        connection.post("#{v1_apps_path(app_id)}/releases/#{release_id}/notes", payload.to_json) do |request|
+          request.headers["Authorization"] = "Bearer " + auth_token
+        end
+        UI.message("Release notes have been posted.")
+      end
+
       def self.get_upload_token(app_id, binary_path)
         begin
           binary_hash = Digest::SHA256.hexdigest(File.open(binary_path).read)
         rescue Errno::ENOENT
-          UI.crash!(ErrorMessage::APK_NOT_FOUND)
+          UI.crash!(ErrorMessage::APK_NOT_FOUND + binary_path)
         end
 
         begin
-          response = connection.get("#{PATH}#{app_id}") do |request|
+          response = connection.get(v1_apps_path(app_id).to_s) do |request|
             request.headers["Authorization"] = "Bearer " + auth_token
           end
         rescue Faraday::ResourceNotFound
-          UI.crash!(ErrorMessage::INVALID_APP_ID)
+          UI.crash!(ErrorMessage::INVALID_APP_ID + app_id)
         end
 
         contact_email = response.body[:contactEmail]
         if contact_email.nil? || contact_email.strip.empty?
-          UI.crash!(ErrorMessage::GET_APP_NO_CONTACT_EMAIL_ERROR)
+          UI.crash!(ErrorMessage::GET_APP_NO_CONTACT_EMAIL_ERROR + "\"#{contact_email}\"")
```

Comment:
I don't see any point in printing out the contact email if it's nil or empty.

Reply:
https://github.com/fastlane/fastlane-plugin-firebase_app_distribution/pull/60#discussion_r447722015  I believe we were addressing this comment by Josh. He states we should print the original contact email to disambiguate if it was an empty contact email or if it was actually not passed in.

Reply:
Yeah NBD on this -- Lee's probably right in that if it's an empty string (probably weird) it will be indistinguishable from nil in the output.